### ISSUE TITLE
confidence intervals for `garch` models are missing

### ISSUE THREAD

**Issue by IndrajeetPatil** [tidymodels/broom#568]
Confidence intervals are available for `garch` models, but they are missing from its respective `broom` tidier:
``` r
# setup
set.seed(123)
library(tseries)
#> Registered S3 method overwritten by 'xts':
#>   method     from
#>   as.zoo.xts zoo
#> Registered S3 method overwritten by 'quantmod':
#>   method            from
#>   as.zoo.data.frame zoo
data(EuStockMarkets)
dax <- diff(log(EuStockMarkets))[,"DAX"]
dax.garch <- tseries::garch(dax, trace = FALSE)

# summary
summary(dax.garch)
#> 
#> Call:
#> tseries::garch(x = dax, trace = FALSE)
#> 
#> Model:
#> GARCH(1,1)
#> 
#> Residuals:
#>       Min        1Q    Median        3Q       Max 
#> -12.18398  -0.47968   0.04949   0.65746   4.48048 
#> 
#> Coefficient(s):
#>     Estimate  Std. Error  t value Pr(>|t|)    
#> a0 4.639e-06   7.560e-07    6.137 8.42e-10 ***
#> a1 6.833e-02   1.125e-02    6.073 1.25e-09 ***
#> b1 8.891e-01   1.652e-02   53.817  < 2e-16 ***
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> 
#> Diagnostic Tests:
#>  Jarque Bera Test
#> 
#> data:  Residuals
#> X-squared = 12947, df = 2, p-value < 2.2e-16
#> 
#> 
#>  Box-Ljung test
#> 
#> data:  Squared.Residuals
#> X-squared = 0.13566, df = 1, p-value = 0.7126

# confidence intervals
confint(dax.garch)
#>           2.5 %       97.5 %
#> a0 3.157588e-06 6.120989e-06
#> a1 4.627785e-02 9.037965e-02
#> b1 8.566877e-01 9.214456e-01

# tidier output (no confidence intervals present)
broom::tidy(x = dax.garch, conf.int = TRUE)
#> # A tibble: 3 x 5
#>   term    estimate   std.error statistic  p.value
#>   <chr>      <dbl>       <dbl>     <dbl>    <dbl>
#> 1 a0    0.00000464 0.000000756      6.14 8.42e-10
#> 2 a1    0.0683     0.0113           6.07 1.25e- 9
#> 3 b1    0.889      0.0165          53.8  0.
```

<sup>Created on 2019-01-03 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ----------------------------------------------------------
#>  setting  value                                             
#>  version  R Under development (unstable) (2018-11-30 r75724)
#>  os       Windows 10 x64                                    
#>  system   x86_64, mingw32                                   
#>  ui       RTerm                                             
#>  language (EN)                                              
#>  collate  English_United States.1252                        
#>  ctype    English_United States.1252                        
#>  tz       Asia/Calcutta                                     
#>  date     2019-01-03                                        
#> 
#> - Packages --------------------------------------------------------------
#>  package     * version     date       lib
#>  assertthat    0.2.0       2017-04-11 [1]
#>  backports     1.1.3       2018-12-14 [1]
#>  bindr         0.1.1       2018-03-13 [1]
#>  bindrcpp      0.2.2       2018-03-29 [1]
#>  broom         0.5.1.9000  2018-12-24 [1]
#>  callr         3.1.1       2018-12-21 [1]
#>  cli           1.0.1.9000  2018-10-30 [1]
#>  crayon        1.3.4       2017-09-16 [1]
#>  curl          3.2         2018-03-28 [1]
#>  desc          1.2.0       2018-10-30 [1]
#>  devtools      2.0.1       2018-10-26 [1]
#>  digest        0.6.18      2018-10-10 [1]
#>  dplyr         0.7.8       2018-11-10 [1]
#>  evaluate      0.12        2018-10-09 [1]
#>  fansi         0.4.0       2018-11-05 [1]
#>  fs            1.2.6       2018-08-23 [1]
#>  generics      0.0.2       2018-11-29 [1]
#>  glue          1.3.0       2018-07-17 [1]
#>  highr         0.7         2018-06-09 [1]
#>  htmltools     0.3.6       2017-04-28 [1]
#>  knitr         1.21        2018-12-10 [1]
#>  lattice       0.20-38     2018-11-04 [2]
#>  magrittr      1.5         2014-11-22 [1]
#>  memoise       1.1.0       2017-04-21 [1]
#>  pillar        1.3.1       2018-12-15 [1]
#>  pkgbuild      1.0.2       2018-10-16 [1]
#>  pkgconfig     2.0.2       2018-08-16 [1]
#>  pkgload       1.0.2       2018-10-29 [1]
#>  prettyunits   1.0.2       2015-07-13 [1]
#>  processx      3.2.1       2018-12-05 [1]
#>  ps            1.3.0       2018-12-21 [1]
#>  purrr         0.2.5.9000  2018-11-28 [1]
#>  quadprog      1.5-5       2013-04-17 [1]
#>  quantmod      0.4-13      2018-04-13 [1]
#>  R6            2.3.0       2018-10-04 [1]
#>  Rcpp          1.0.0       2018-11-07 [1]
#>  remotes       2.0.2       2018-10-30 [1]
#>  rlang         0.3.0.1     2018-10-25 [1]
#>  rmarkdown     1.11        2018-12-08 [1]
#>  rprojroot     1.3-2       2018-01-03 [1]
#>  sessioninfo   1.1.1       2018-11-05 [1]
#>  stringi       1.2.4       2018-07-20 [1]
#>  stringr       1.3.1       2018-05-10 [1]
#>  testthat      2.0.1       2018-10-13 [1]
#>  tibble        1.4.99.9006 2018-12-21 [1]
#>  tidyr         0.8.2       2018-10-28 [1]
#>  tidyselect    0.2.5       2018-10-11 [1]
#>  tseries     * 0.10-46     2018-11-19 [1]
#>  TTR           0.23-4      2018-09-20 [1]
#>  usethis       1.4.0.9000  2018-12-12 [1]
#>  utf8          1.1.4       2018-05-24 [1]
#>  withr         2.1.2       2018-03-15 [1]
#>  xfun          0.4         2018-10-23 [1]
#>  xts           0.11-2      2018-11-05 [1]
#>  yaml          2.2.0       2018-07-25 [1]
#>  zoo           1.8-4       2018-09-19 [1]
#>  source                           
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  Github (tidymodels/broom@a1ac666)
#>  CRAN (R 3.6.0)                   
#>  Github (r-lib/cli@56538e3)       
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  Github (r-lib/desc@7c12d36)      
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  Github (brodieG/fansi@ab11e9c)   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.6.0)                   
#>  Github (tidyverse/purrr@b4ae036) 
#>  CRAN (R 3.5.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  Github (tidyverse/tibble@f1eacb5)
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  Github (r-lib/usethis@923dd75)   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.6.0)                   
#>  CRAN (R 3.5.1)                   
#>  CRAN (R 3.5.1)                   
#> 
#> [1] C:/Users/inp099/Documents/R/win-library/3.6
#> [2] C:/Program Files/R/R-devel/library
```

</details>

**Comment by IndrajeetPatil** [comment]
@alexpghayes This is also a beginner-friendly issue.

**Comment by petrhrobar** [comment]
Hi There,
I would like to start contributing and help update the functions, this issue is tagged as beginer friendly. I have already  helped with basic function for sarlm objects here (https://github.com/tidymodels/broom/pull/847). I would like to start helping I really liked writting tidy functions!

**Comment by simonpcouch** [comment]
Hey @petrhrobar! You'll want to fork broom's `master` branch and then take a look at the `tidy.garch` definition in `R/tseries-tidiers.R`. You'll need to add `conf.int` and `conf.level` arguments, implement them (probably using the [`conf.int` idiom](https://github.com/tidymodels/broom/blob/master/R/spdep-tidiers.R#L80-L84)), document them with `devtools::document()`, and then commit and push to your fork. From there, you can pull request back to broom's `master`! Make sure you add yourself to the `DESCRIPTION` and note the change in `NEWS.md`, too.

**Comment by simonpcouch** [comment]
Closed in #964. Thanks for yalls work and patience here!
